cfn-resources/access-list-api-key/cmd/resource/resource.go

```diff
@@ -227,6 +214,23 @@ func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler
 		ResourceModel:   nil}, nil
 }
 
+// If CIDR block is defined, subnet mask is removed so get and delete requests work correctly
```

Comment:
fix inspired from https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/accesslistapikey/resource_access_list_api_key.go#L84. Also reached out to IAM team to confirm the current behavior (or limitation) of [getApiKeyAccessList](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/v2/#tag/Programmatic-API-Keys/operation/getApiKeyAccessList) which cannot receive the cidr block value directly.

Reply:
I think this is not the same implementation.  In TF code, the switch only enters one path, so it's like using an if else, not 2 continuous if statements.

also note that the TF code could be quite simplified to: 
```

if  CIDRBlock != "" {
		parts := strings.SplitN(CIDRBlock, "/", 2)
		if parts[1] == "32" {
			entry = parts[0]
		} else {
			entry = CIDRBlock
		}
} else {
		entry = IPAddress
}
```
i think we could use that idea here as well

Reply:
can CidrBlock and IPAddress exist at the same time?

Reply:
They cannot, it is being validated in the create [here](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/access-list-api-key/cmd/resource/resource.go#L84). One must be defined, and both cannot be defined at the same time. Having an else should be the same as checking that ip address is defined.

Reply:
nice, thx